include/mqtt/callable_overlay.hpp

```diff
@@ -623,20 +623,24 @@ struct callable_overlay final : public Impl
      *
      * This handler is called if the client called `disconnect()` and the server closed the socket cleanly.
      * If the socket is closed by other reasons, error_handler is called.
+     * This calls base class on_close().
```

Comment:
Probably should say

"This calls base::on_close() prior to calling the provided callback, if any"

Reply:
Thanks, I will update the doc.